### PR TITLE
Adding Grid.tick_params() method.

### PR DESCRIPTION
**Pull request by stefmolin**
Closes https://github.com/mwaskom/seaborn/issues/2491

`FacetGrid` and `PairGrid` now have a `tick_params()` method, which will iterate over each `Axes` in the grid and call its `tick_params()` method. Some examples:

```python
import seaborn as sns

tips = sns.load_dataset('tips')

g = sns.FacetGrid(tips, col='time', row='sex')
g.map(sns.scatterplot, 'total_bill', 'tip')
g.tick_params(direction='in', length=6, width=2) # on both
g.tick_params(colors='blue', axis='x')
g.tick_params(colors='green', axis='y')
```
Screen Shot 2022-08-07 at 10 51 44 AM

```python
import seaborn as sns

penguins = sns.load_dataset('penguins')

g = sns.PairGrid(penguins, diag_sharey=False, corner=True)
g.map_lower(sns.scatterplot)
g.map_diag(sns.kdeplot)
g.tick_params(direction='in', length=6, width=2, color='blue')
```
Screen Shot 2022-08-06 at 8 47 55 PM

### CODE REVIEW

**Comment by stefmolin** [comment]
I wasn't sure if you wanted to put this in the 0.12.0 release or the 0.13.0 release, so I held off on adding to the What's New for now. I also didn't add to the examples in case you wanted to either leave it out or had a specific change in mind. Let me know what you're thinking.

File: seaborn/axisgrid.py [hunk 1]
```diff
@@ -278,6 +278,24 @@ def legend(self):
         except AttributeError:
             return None
 
+    def tick_params(self, **kwargs):
```
Comment (mwaskom) [inline seaborn/axisgrid.py:281]: Can we accept `axis` as a positional parameter like `Axes.tick_params` does?

File: seaborn/axisgrid.py [hunk 2]
```diff
@@ -278,6 +278,24 @@ def legend(self):
...
+        """Modify the ticks, tick labels, and gridlines.
+
+        Parameters
+        ----------
+        **kwargs
```
Comment (mwaskom) [inline seaborn/axisgrid.py:286]: I think the `**` is going to mess up numpydoc formatting (I haven't build the docs to verify though).

File: tests/test_axisgrid.py [hunk 3]
```diff
@@ -696,6 +696,19 @@ def f(grid, color):
...
+    def test_tick_params(self):
+
+        g = ag.FacetGrid(self.df, row="a", col="b")
+        color = 'blue'
+        width = 2
+        g.tick_params(width=width, color=color)
+        for ax in g.axes.ravel():
+            for axis in ['xaxis', 'yaxis']:
+                for tick in getattr(ax, axis).get_major_ticks():
+                    assert tick.tick1line.get_color() == color
```
Comment (mwaskom) [inline tests/test_axisgrid.py:708]: Nit, using `assert mpl.colors.same_color(tick.tick1line.get_color(), color)` would make this more robust if matplotlib coerces the internal color representation to rgba in the future.

File: tests/test_axisgrid.py [hunk 4]
```diff
@@ -696,6 +696,19 @@ def f(grid, color):
         assert res == color
         assert g.figure.get_facecolor() == color
 
+    def test_tick_params(self):
+
+        g = ag.FacetGrid(self.df, row="a", col="b")
+        color = 'blue'
+        width = 2
+        g.tick_params(width=width, color=color)
+        for ax in g.axes.ravel():
```
Comment (mwaskom) [inline tests/test_axisgrid.py:705]: Super nit (more of an FYI): using `array.flat` is more idiomatic than `array.ravel()` for iteration.

**Review by mwaskom — CHANGES_REQUESTED** [review]
Thanks @stefmolin, looks great, just a couple tiny comments.

**Comment by mwaskom** [comment]
> I wasn't sure if you wanted to put this in the 0.12.0 release or the 0.13.0 release, so I held off on adding to the What's New for now.

This can go into 0.12 as I haven't cut a release candidate yet, so please update the release notes.

**Comment by stefmolin** [comment]
Comments addressed and release notes updated. I also updated the usage of `ravel()` in the tests from my previous PR for `refline()`.

File: tests/test_axisgrid.py [hunk 5]
```diff
@@ -696,6 +696,19 @@ def f(grid, color):
...
+        g = ag.FacetGrid(self.df, row="a", col="b")
+        color = 'blue'
+        width = 2
+        g.tick_params(width=width, color=color)
+        for ax in g.axes.flat:
+            for axis in ['xaxis', 'yaxis']:
+                for tick in getattr(ax, axis).get_major_ticks():
+                    assert mpl.colors.same_color(tick.tick1line.get_color(), color)
+                    assert mpl.colors.same_color(tick.tick2line.get_color(), color)
+                    assert tick._width == width
```
Comment (mwaskom) [inline tests/test_axisgrid.py:710]:
```suggestion
                    assert tick.tick1line.get_linewidth() == width
```

File: tests/test_axisgrid.py [hunk 6]
```diff
@@ -1431,6 +1444,19 @@ def test_legend(self):
...
+        g = ag.PairGrid(self.df)
+        color = 'red'
+        width = 3
+        g.tick_params(width=width, color=color)
+        for ax in g.axes.flat:
+            for axis in ['xaxis', 'yaxis']:
+                for tick in getattr(ax, axis).get_major_ticks():
+                    assert mpl.colors.same_color(tick.tick1line.get_color(), color)
+                    assert mpl.colors.same_color(tick.tick2line.get_color(), color)
+                    assert tick._width == width
```
Comment (mwaskom) [inline tests/test_axisgrid.py:1458]:
```suggestion
                    assert tick.tick1line.get_linewidth() == width
```

**Review by mwaskom — CHANGES_REQUESTED** [review]
Brilliant, thanks @stefmolin! Last bit, and sorry for missing this in my first review, but we should avoid accessing private matplotlib attributes as a general rule. I think the suggested change will work, and then this is 🟢🟢🟢

**Comment by mwaskom** [comment]
Odd, I wonder why that didn't work.

Since matplotlib seems to be doing something weird here, probably best to use a different property. `pad` seems to work:

```python
ax = plt.gca()
ax.tick_params(pad=4.7)
yt, *_ = ax.yaxis.get_major_ticks()
print(yt.get_pad())
```
4.7

Feel free to change locally and force push over my attempt at a fix.

**Comment by stefmolin** [comment]
Switched to `pad`, as suggested.

**Review by mwaskom — APPROVED** [review]

**Comment by mwaskom** [comment]
Wonderful!

**Pull request merged**